core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -101,13 +103,20 @@ Object executeAndDecode(RequestTemplate template) throws Throwable {
     }
     long elapsedTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
 
+    boolean shouldClose = true;
     try {
       if (logLevel != Logger.Level.NONE) {
         response =
             logger.logAndRebufferResponse(metadata.configKey(), logLevel, response, elapsedTime);
       }
       if (Response.class == metadata.returnType()) {
-        if (response.body() == null) {
+        if (response.body() == null || response.body().isRepeatable()) {
+          return response;
+        }
+        if (response.body().length() == null ||
+                response.body().length() > MAX_RESPONSE_BUFFER_SIZE ||
+                response.body().length() < 0) {
```

Comment:
Good call, I have not seen a negative value there.
I also think that checking isRepeatable() is wrong in the above condition, it may have to get resolved before being closed.  I'll remove that part of the conditional.